chain/jsonrpc/client/src/lib.rs

```diff
@@ -1,54 +1,58 @@
+#![feature(trait_alias)]
 use std::time::Duration;
 
 use actix_web::client::Client;
 use futures::Future;
 use serde::Serialize;
 
-use near_chain::Block;
-use near_client::StatusResponse;
-use near_primitives::rpc::QueryResponse;
+use near_primitives::block::Block;
+use near_primitives::rpc::{QueryResponse, StatusResponse};
 use near_primitives::transaction::{FinalTransactionResult, TransactionResult};
 use near_primitives::types::BlockIndex;
 
+pub mod message;
 use crate::message::{from_slice, Message};
 
 /// Timeout for establishing connection.
 const CONNECT_TIMEOUT: Duration = Duration::from_secs(10);
 
-type HttpRequest<T> = Box<dyn Future<Item = T, Error = String>>;
-type RpcRequest<T> = Box<dyn Future<Item = T, Error = String>>;
+pub trait HttpRequest<T> = Future<Item = T, Error = String>;
```

Comment:
Ideally we should avoid using features that are nightly only unless really necessary.